### PR TITLE
Fix parallel example exception

### PR DESCRIPTION
**Pull request by Ayaro**
Fix for:
1) Class prediction in parallel ensembles
2) Different number of samples in parallel ensambles

### CODE REVIEW

File: example/example_parallel.py [hunk 1]
```diff
@@ -51,7 +54,7 @@ def spawn_classifier(seed, dataset_name):
     
     digits = sklearn.datasets.load_digits()
     X = digits.data
-    y = digits.target
+    y = digits.target + 1
```
Comment (mfeurer) [inline example/example_parallel.py]: Why do you increase the index by one in the example?

File: example/example_parallel.py [hunk 2]
```diff
@@ -8,10 +8,13 @@
 
 from autosklearn.classification import AutoSklearnClassifier
 from autosklearn.constants import *
+from autosklearn.util.backend import BackendContext
 
 tmp_folder = '/tmp/autosklearn_parallel_example_tmp'
 output_folder = '/tmp/autosklearn_parallel_example_out'
 
+backend_context = BackendContext(tmp_folder, output_folder, True, True)
```
Comment (mfeurer) [inline example/example_parallel.py]: I think the backend should not be used in the examples. Can you replace this by a simple os.remove?

File: autosklearn/estimators.py [hunk 3]
```diff
@@ -253,11 +253,11 @@ def fit(self, *args, **kwargs):
         self._automl = self.build_automl()
         super(AutoSklearnEstimator, self).fit(*args, **kwargs)
 
-    def fit_ensemble(self, task=None, metric=None, precision='32',
+    def fit_ensemble(self, y, task=None, metric=None, precision='32',
```
Comment (mfeurer) [inline autosklearn/estimators.py]: Don't you have to change the signature also in the AutoMLDecorator?

File: test/test_automl/test_estimators.py [hunk 4]
```diff
@@ -93,12 +95,13 @@ def test_feat_type_wrong_arguments(self):
                                 cls.fit,
                                 X=X, y=y, feat_type=['Car']*100)
 
-    @unittest.skip("pSMAC not yet working with new python SMAC")
     def test_fit_pSMAC(self):
         output = os.path.join(self.test_dir, '..', '.tmp_estimator_fit_pSMAC')
         self._setUp(output)
 
         X_train, Y_train, X_test, Y_test = putil.get_dataset('iris')
+        Y_train = Y_train + 1
```
Comment (mfeurer) [inline test/test_automl/test_estimators.py]: Could you please add a comment why the labels are increased by one?

File: test/test_automl/test_estimators.py [hunk 5]
```diff
@@ -142,10 +146,20 @@ def test_fit_pSMAC(self):
...
+                            metric=ACC_METRIC,
+                            precision='32',
+                            dataset_name='iris',
+                            ensemble_size=20,
+                            ensemble_nbest=50)
+
+        predictions = automl.predict(X_test)
+        score = sklearn.metrics.accuracy_score(Y_test, predictions)
+
+        # TODO: score calculation fails
```
Comment (mfeurer) [inline test/test_automl/test_estimators.py]: Can you remove the TODO?

File: test/test_automl/test_estimators.py [hunk 6]
```diff
@@ -142,10 +146,20 @@ def test_fit_pSMAC(self):
...
+        # internal_score = automl.score(X_test, Y_test)
+        # self.assertEqual(score, internal_score)
```
Comment (mfeurer) [inline test/test_automl/test_estimators.py]: Could you please add a check that the dummy model is actually included in the ensemble?

File: autosklearn/estimators.py (hunk 3 above)
Comment (Ayaro) [inline autosklearn/estimators.py]: Well, I'd say rather not - we don't need to change the signature for now for any other method in AutoML and Regressor

Comment (mfeurer) [inline autosklearn/estimators.py]: I see your point. After thinking about this for a little bit, I think it's more important to have the interface for regression and classification as similar as possible. Even if this implies that we have to pass an unused `y` to `fit_ensemble`. Do you see any reason why this is bad?

Comment (Ayaro) [inline autosklearn/estimators.py]: Well, I'd personally even use the `X` parameter too, I'd say this signature would be even more convenient (for all the estimators). IMO the only negative side is the force to use some parameters that doesn't matter and do not override the data from `datamanager`.
